src/training/data.py

```diff
@@ -46,24 +52,41 @@ def __getitem__(self, idx):
         texts = tokenize([str(self.captions[idx])])[0]
         return images, texts
 
+
 @dataclass
 class DataInfo:
     dataloader: DataLoader
     sampler: DistributedSampler
 
+
 def preprocess_txt(text):
     return tokenize([str(text)])[0]
 
+
 def get_dataset_size(shards):
     shards_list = list(braceexpand.braceexpand(shards))
     dir_path = os.path.dirname(shards)
-    sizes_filename = os.path.join(dir_path, 'sizes.json')
-    sizes = json.load(open(sizes_filename, 'r'))
-    total_size = sum(
-        [int(sizes[os.path.basename(shard)]) for shard in shards_list])
+    if 'sizes.json' in os.listdir(dir_path):
+        sizes_filename = os.path.join(dir_path, 'sizes.json')
+        sizes = json.load(open(sizes_filename, 'r'))
+        total_size = sum(
+            [int(sizes[os.path.basename(shard)]) for shard in shards_list])
+    elif '__len__' in os.listdir(dir_path):
+        total_size = eval(open(os.path.join(dir_path, '__len__'), 'r').read())
+    else:
+        # name / path based hacks (NOTE: specific to a given download / instances of dataset in terms of samples)
+        if 'cc3m-train' in shards:
```

Comment:
Can we print to the logs when this happens? It would be good to make sure the codebase is not silently doing something users don't want to. For instance, if someone is using a filtered or a bigger version of the laion dataset that contains "laion" in it, using the value 407332084 would be undesirable

Reply:
thanks @rwightman, adding a flag sounds good to me